doc/static_route/SONiC_static_route_hdl.md

```diff
@@ -0,0 +1,546 @@
+|create route for a prefix with NH IP addresses and distance|    verify the route is created|
+|create route for a prefix with NH IP address, interface and distance | verify route is created|
+|create route for a prefix with interface and distance | verify route is created|
+|create route for a prefix with NH IP addresses, NH VRF and distane |    verify the route is created|
+|create route for a prefix with NH IP address, interface and distance| verify route is created|
+|create route for a prefix with interface, NH VRF and distance | verify route is created|
+|create route for a prefix with blackhole and distance |verify route is created
+|CLI: show ip route static|   verify static routes are displayed with flag "S" (i.e Static)|
+|Delete route for a prefix with NH IP addresses| verify the route is deleted|
+|Delete route for a prefix with NH IP address and interface| verify route is deleted|
+|Delete route for a prefix with interface only| verify route is deleted|
+|Delete route for a prefix with NH IP addresses and NH VRF|    verify the route is deleted|
+|Delete route for a prefix with NH IP address, interface and NH VRF| verify route is deleted|
+|Delete route for a prefix with interface and NH VRF| verify route is deleted|
+|Delete route for a prefix with blackhole| verify route is deleted|
+
+| Test cases for management static routes |Test Result |
+| :------ | :----- |
+|create route in default instance  for a prefix with NH IP address and interface as Management0 | verify route is created|
+|create route in default instance  for a prefix with interface as Management0 | verify route is created|
+|create route in mgmt vrf  for a prefix with NH IP address and interface as in-band interface | verify route is created|
+|create route in mgmt vrf  for a prefix with NH IP address | verify route is created|
+|create route in mgmt vrf  for a prefix with in-band interface | verify route is created|
+|create route in data VRF for a prefix with NH IP addresses and NH VRF as management vrf|    verify the route is created|
+|create route in data VRF for a prefix with NH interface and NH VRF as management vrf|    verify the route is created|
+|create route in data VRF for a prefix with NH interface as Management0 and NH VRF as default|    verify the route is created|
+|create route in mgmt vrf for a prefix with blackhole and distance |verify route is created
+|CLI: show ip route vrf mgmt static|   verify static routes are displayed with flag "S" (i.e Static)|
+|Delete all the above configured routes| verify all route are deleted
+|CLI: show ip route vrf mgmt static|   verify no static routes that are deleted are displayed|
+
+| Negative test cases |Test Result |
+| :------ | :----- |
+|Try creating a route with mismatched prefix(IPV4) and address(IPV6)| test will fail|
+|Try creating a route with wrong distance like 300| test will fail|
+|Try creating a route with non-existing VRF| backened will fail|
+

```

Comment:
Generate comments
1. Please add test cases to make sure route creation is validated in kernel, APP_DB and HW
2. Please have data forwarding test cases to ensure that data is forwarded across the VRF. That is data is forwarded from mgmt VRF to user/default VRF and vice versa
3. Added a save/reboot test case

Reply:
Will see to add some spytest to check the kernel and HW configs.